### PR TITLE
feat(metrics): Decouple Grafana and Prometheus installation for OSM install command

### PR DESCRIPTION
**Pull request by nshankar13**
Remove the "enableMetricStack" flag for the OSM install command. Replace with two separate flags, enable-grafana and enable-prometheus to decouple installation of Grafana and Prometheus

Please mark with X for applicable areas.

- New Functionality      [ ]
- Documentation          [ ]
- Install                [X]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests / CI System      [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

File: charts/osm/values.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ OpenServiceMesh:
   enableBackpressureExperimental: false
   enableEgress: false
   enableMetricsStack: true
+  enableGrafana: true
```
Comment (snehachhabria) [inline charts/osm/values.yaml:36]: Now that we have a separate flag for grafana the `enableMetricsStack` is primarily used for Prometheus deployment, maybe we should change that flag as well to make it more explicit 

@eduser25 as FYI

Comment (eduser25) [reply charts/osm/values.yaml:36]: +1, agree with Sneha, at this point let's have independant toggles for Grafana / Prometheus deployments

Comment (nshankar13) [reply charts/osm/values.yaml:36]: Ok, sounds good!

File: charts/osm/templates/grafana-configmap.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.OpenServiceMesh.enableMetricsStack }}
+{{- if and .Values.OpenServiceMesh.enableMetricsStack .Values.OpenServiceMesh.enableGrafana}}
```
Comment (snehachhabria) [inline charts/osm/templates/grafana-configmap.yaml:1]:
```suggestion
{{- if and .Values.OpenServiceMesh.enablePrometheus .Values.OpenServiceMesh.enableGrafana}}
```

File: charts/osm/values.yaml [hunk 3]
```diff
@@ -33,6 +33,7 @@ OpenServiceMesh:
   enableBackpressureExperimental: false
   enableEgress: false
   enableMetricsStack: true
```
Comment (snehachhabria) [inline charts/osm/values.yaml:35]: Rename this to enablePrometheus

File: cmd/cli/install.go [hunk 4]
```diff
@@ -94,7 +94,10 @@ type installCmd struct {
 	enableBackpressureExperimental bool
 
 	// Toggle to deploy/not deploy metrics (Promethus+Grafana) stack
```
Comment (ksubrmnn) [inline cmd/cli/install.go:96]: Please update this comment

File: cmd/cli/install_test.go [hunk 5]
```diff
@@ -811,7 +820,169 @@ var _ = Describe("Resolving values for install command with vault parameters", f
...
 				"meshCIDRRanges":                 testMeshCIDR,
-				"enableMetricsStack":             true,
+				"enablePrometheus":               true,
+				"enableGrafana":                  true,
+				"deployJaeger":                   false,
+			}}))
+	})
+})
+
+var _ = Describe("Making sure that grafana is disabled when flag is set to false", func() {
```
Comment (ksubrmnn) [inline cmd/cli/install_test.go:830]: Maybe instead : "Ensure that grafana is disabled when flag is set to false"

File: cmd/cli/install_test.go [hunk 6]
```diff
@@ -811,7 +820,169 @@ var _ = Describe("Resolving values for install command with vault parameters", f
...
+				"meshCIDRRanges":                 testMeshCIDR,
+				"enablePrometheus":               true,
+				"enableGrafana":                  false,
+				"deployJaeger":                   false,
+			}}))
+	})
+
+})
+
+var _ = Describe("Making sure that prometheus is disabled when flag is set to false", func() {
```
Comment (ksubrmnn) [inline cmd/cli/install_test.go:911]: Same here

**Comment by shashankram** [comment]
@nshankar13 Could you kindly update the PR and commit title and description.

**Review by ksubrmnn — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

**Review by snehachhabria — COMMENTED** [review]
Please make the required documentation changes as well ( one such document is observability.md )

**Comment by ksubrmnn** [comment]
Fixes #1599

**Comment by shashankram** [comment]
> Please make the required documentation changes as well ( one such document is observability.md )

@nshankar13 would you mind following this up with the requested documentation changes.

**Pull request merged**